kagiso_smart_404/middleware.py

```diff
@@ -0,0 +1,15 @@
+import logging
+
+from .constants import TRACKING_QUERY_STRING_KEY, TRACKING_QUERY_STRING_VALUE
+
+
+logger = logging.getLogger('django')
+
+
+class SmartRedirectMetricsMiddleware:
+
+    def process_request(self, request):
+        query_string = request.GET.get(TRACKING_QUERY_STRING_KEY)
+
+        if query_string == TRACKING_QUERY_STRING_VALUE:
+            logger.info('count#smart_404.suggested_page.clicked=1')
```

Comment:
So the name of the metric isn't completely accurate as the tracking_query_string will be appended to the automatic redirect as well, so that wouldn't be measuring if people are just clicking it.